baseline/baseline.py

```diff
@@ -0,0 +1,195 @@
+"""
+Baseline model for HEAR 2021 NeurIPS competition.
+
+This is simply a mel spectrogram followed by random projection.
+"""
+
+import math
+from collections import defaultdict
+from typing import Any, Dict, List, Optional, Tuple
+
+import torch
+from torch import Tensor
+from torchaudio.transforms import MelSpectrogram
+
+
+class RandomProjectionMelEmbedding(torch.nn.Module):
+    n_fft = 4096
+    n_mels = 256
+    sample_rate = 44100
+    seed = 0
+
+    def __init__(self):
+        super().__init__()
+        torch.random.manual_seed(self.seed)
+        self.mel = MelSpectrogram(
+            sample_rate=self.sample_rate, n_fft=self.n_fft, n_mels=self.n_mels
+        )
+        self.emb4096 = torch.nn.Parameter(
+            torch.rand(self.n_mels, 4096) / math.sqrt(self.n_mels)
+        )
+        self.emb2048 = torch.nn.Parameter(torch.rand(4096, 2048) / math.sqrt(4096))
+        self.emb512 = torch.nn.Parameter(torch.rand(2048, 512) / math.sqrt(2048))
+        self.emb128 = torch.nn.Parameter(torch.rand(512, 128) / math.sqrt(512))
+        self.emb20 = torch.nn.Parameter(torch.rand(128, 20) / math.sqrt(128))
+        self.activation = torch.nn.Sigmoid()
+
+    def forward(self, x: Tensor, hop_size_samples: int, center: bool):
+        self.mel.hop_length = hop_size_samples
+        self.mel.center = center
+        self.mel.spectrogram.hop_length = hop_size_samples
+        self.mel.spectrogram.center = center
+        x = torch.log(self.mel(x) + 1e-4)
```

Comment:
This would be a good candidate for a class variable: `eps = 1e-4` or some such thing. Adjustable, but a just a little out of reach.